File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.py
import itertools

def isPrime(num):
    result = 0
    if num == 1:
        return False
    elif num == 2:
        return True
    for i in range(2, int(num**0.5) + 1):
        if num % i == 0:
            return False
    return True

def solution(nums):
    answer = 0
    
    nums = list(itertools.combinations(nums, 3))
    
    for num in nums:
        if isPrime(sum(num)):
            answer += 1

    return answer

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.js
function solution(nums) {
    
    const isPrimeNumber = (num) => {
        if (num === 1) return false;
        let result = 0;
        for (let i = 1; i <= num; i++) {
            if (num % i === 0) result++;
            
            if (result > 2) return false;
        }
        return true;
    }
    
    
    let answer = 0;
    // dfs로 구현
    const combinations = (currentArray, startIndex) => {
        // 종료 조건
        if (currentArray.length === 3) {
            if (isPrimeNumber(currentArray.reduce((sum, num) => sum + num))) {
                answer += 1;
            }
            return;
        }
        
        for (let i = startIndex; i < nums.length; i++) {
            combinations([...currentArray, nums[i]], i + 1);
        }
    }
    
    combinations([], 0);
    
    return answer;
}